File: src/presentation/http/dtos/user_dtos.rs
use uuid::Uuid;

use crate::domain::User;

#[derive(serde::Serialize)]
pub struct UserResponse {
    pub id: Uuid,
    pub email: String,
    pub full_name: String,
    pub created_at: chrono::NaiveDateTime,
    pub updated_at: chrono::NaiveDateTime,
}

impl From<User> for UserResponse {
    fn from(user: User) -> Self {
        Self {
            id: user.id,
            email: user.email,
            full_name: user.full_name,
            created_at: user.created_at.naive_utc(),
            updated_at: user.updated_at.naive_utc(),
        }
    }
}

File: internal/presentation/rest/dtos/user_dtos.go
package dtos

import (
	"time"

	"github.com/yelaco/ai-document-backend/internal/domain/models/entity"
)

type UserResponse struct {
	ID        string `json:"id"`
	Email     string `json:"email"`
	FullName  string `json:"full_name"`
	Role      string `json:"role"`
	CreatedAt string `json:"created_at"`
}

func UserResponseFromEntity(user *entity.User) *UserResponse {
	return &UserResponse{
		ID:        user.ID.String(),
		Email:     user.Email,
		FullName:  user.FullName,
		Role:      string(user.Role),
		CreatedAt: user.CreatedAt.Format(time.RFC3339),
	}
}
